### ISSUE TITLE
`required` function swallows message until first colon (`:`)

### ISSUE THREAD

**Issue by AndiDog** [helm/helm#6044]
Output of `helm version`:

```
version.BuildInfo{Version:"v3.0.0-alpha.1", GitCommit:"b9a54967f838723fe241172a6b94d18caf8bcdca", GitTreeState:"clean"}
```

Example issue (note escaping of `$` is for the shell):

```
export PATH="$HOME/bin/helm-v3.0.0-alpha.1-darwin-amd64:$PATH"
cd /tmp
rm -rf colon-missing
helm create colon-missing
cat >colon-missing/templates/deployment.yaml <<EOF
{{ \$someEmptyValue := "" }}
{{ \$myvar := "abc" }}
{{- required (printf "%s: something is missing" \$myvar) \$someEmptyValue | repeat 0 }}
EOF
helm template myrelease colon-missing/
```

This will fail with a truncated error message:

```
Error: render error at (colon-missing/templates/deployment.yaml:3:4): something is missing
```

Once you remove the `:` from the custom error message (`printf "%s <colon removed here> something is missing" $myvar`), the error output is as expected:

```
Error: render error at (colon-missing/templates/deployment.yaml:3:4): abc <colon removed here> something is missing
```

**Comment by AndiDog** [comment]
This code looks really suspicious:

```
func parseTemplateError(filename string, err error) error {
	tokens := strings.Split(err.Error(), ": ")
	if len(tokens) == 1 {
		// This might happen if a non-templating error occurs
		return fmt.Errorf("render error in (%s): %s", filename, err)
	}
	// The first token is "template"
	// The second token is either "filename:lineno" or "filename:lineNo:columnNo"
	location := tokens[1]
	// The remaining tokens make up a stacktrace-like chain, ending with the relevant error
	errMsg := tokens[len(tokens)-1]
	return fmt.Errorf("render error at (%s): %s", string(location), errMsg)
}
```

**Comment by bacongobbler** [comment]
@AndiDog feel free to contribute a fix if you manage to determine the issue.

**Comment by bacongobbler** [comment]
cc @ian-howell. Would you mind taking a look into this?

**Comment by ian-howell** [comment]
It looks like this happens when there's an instance of `": "` in the important bit of the error message. I'll take a closer look and get a fix in

**Comment by ian-howell** [comment]
So the `text/template` package does some pretty nasty things when it runs into an error while running a function. For example, this template:

```
{{- $foo := "" -}}
{{required "foo is required" $foo}}
```

creates the following error:
```
template: requiredTest:3:2: executing "requiredTest" at <required "foo is required" $foo>: error calling required: foo is required
```

That message has a ton of redundancy, but unfortunately no way to parse out the important bits without possibly treading on a user's toes (e.g. parsing on colons).

I've submitted [a Pull Request](https://github.com/helm/helm/pull/6066) which addresses this by inserting arbitrary delimiters around the user's warning message, then parsing out the message if an error occurs. The PR includes a unit test that addresses this specific issue.

**Comment by thomastaylor312** [comment]
Resolved in #6066
